### PR TITLE
[JBPM-9657] Job execution retryDelay not honored after LoadAndScheduleRequestsTask

### PR DESCRIPTION
**Pull request by elguardian**
Jira : https://issues.redhat.com/browse/JBPM-9657

### CODE REVIEW

**Review by gmunozfe — COMMENTED** [review]
Works fine, but the scheduled map belonging to `PrioritisedScheduledThreadPoolExecutor` doesn't remove error entries (after all retries), so in the long term, it might increase its size without purging these error entries.

File: jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/AbstractAvailableJobsExecutor.java [hunk 1]
```diff
@@ -183,19 +183,17 @@ public void executeGivenJob(RequestInfo request) {
...
                     if (callbacks == null) {
                         callbacks = classCacheManager.buildCommandCallback(ctx, cl);
                     }
-                    
-                	processReoccurring = handleException(request, e, ctx, callbacks);
-                	
+
+                    processReoccurring = handleException(request, e, ctx, callbacks);
                 } finally {
-                    ((ExecutorImpl) executor).clearExecution(request.getId());
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/AbstractAvailableJobsExecutor.java:198]: The clean up must happen after the retries has been exhausted and that is the bug in here. check handleException once the PR is updated.

**Review by gmunozfe — APPROVED** [review]
Looks good to me, fabulous work @elguardian !

**Comment by elguardian** [comment]
Jenkins retest this

**Comment by elguardian** [comment]
Jenkins retest this

**Pull request merged**